assets/apps/components/src/Controls/FormTokenFieldControl.js

```diff
@@ -0,0 +1,89 @@
+import PropTypes from 'prop-types';
+import { FormTokenField } from '@wordpress/components';
+
+const FormTokenFieldControl = ({
+	label,
+	description,
+	choices,
+	onChange,
+	value,
+}) => {
+	const updateValue = (nextVal) => {
+		// Convert option labels into ids
+		const selectedOptionsArray = [];
+
+		for (const optionName of nextVal) {
+			const matchingPost = choices.find((option) => {
+				return option.label === optionName;
+			});
+			if (matchingPost !== undefined) {
+				selectedOptionsArray.push(matchingPost.id);
+			}
+		}
+
+		onChange(selectedOptionsArray);
+	};
+
+	const getValueKeys = () => {
+		// Convert option ids into labels
+		const valueKeys = [];
+		for (const optionId of value) {
+			const wantedPost = choices.find((option) => {
+				return option.id === optionId;
+			});
+			if (wantedPost === undefined || !wantedPost) {
+				continue;
+			}
+			valueKeys.push(wantedPost.label);
+		}
+		return valueKeys;
+	};
+
+	const getOptionNames = () => {
+		return choices.map((option) => option.label);
+	};
+
+	const hasSubArray = (master, sub) => {
+		return sub.every(
+			(
+				(i) => (v) =>
+					(i = master.indexOf(v, i) + 1)
+			)(0)
+		);
+	};
```

Comment:
I think there is an easier way to determine if an array is a sub-array of another, also this implementation is not very eligible. We can also have this as a utility function that you can use in other components.

Reply:
Thank you for the review, I've changed the function to something simpler `sub.every((e) => master.includes(e));`. Didn't add it as a utility function as it's a one-liner and really easy to check.